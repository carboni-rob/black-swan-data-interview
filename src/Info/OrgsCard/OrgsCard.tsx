import React from "react";

type Props = {
  orgs: any[];
};

const OrgsCard = ({ orgs }: Props): JSX.Element => {
  return (
    <div className="card">
      <h2>User Organizations: {orgs.length}</h2>
      <div className="cardContent">
        {orgs.map(
          (org: any): JSX.Element => {
            return (
              <a href={org.url} key={org.id}>
                <img alt="" src={org.avatar_url} className="orgAvatar" />
                {org.login}
              </a>
            );
          }
        )}
      </div>
    </div>
  );
};

export default OrgsCard;
