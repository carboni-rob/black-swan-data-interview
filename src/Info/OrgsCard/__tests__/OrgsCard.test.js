/* eslint-disable @typescript-eslint/camelcase */
import React from "react";
import { render } from "@testing-library/react";
import OrgsCard from "../OrgsCard";

describe("<OrgsCard />", () => {
  const orgs = [
    {
      id: 1,
      login: "test1",
      url: "test1/orgs/url",
      avatar_url: "test1/avatar/url"
    },
    {
      id: 2,
      login: "test2",
      url: "test2/orgs/url",
      avatar_url: "test2/avatar/url"
    },
    {
      id: 3,
      login: "test3",
      url: "test3/orgs/url",
      avatar_url: "test3/avatar/url"
    }
  ];

  const tree = <OrgsCard orgs={orgs} />;

  it("should match last snapshot", () => {
    const { container } = render(tree);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should contain a title with the number of organizations", () => {
    const { getByText } = render(tree);

    getByText("User Organizations: 3");
  });

  it("should contain a link with appropriate href for each element", () => {
    const { getByText } = render(tree);

    orgs.map(org => {
      expect(getByText(org.login)).toHaveProperty(
        "href",
        `http://localhost/${org.url}`
      );

      return false;
    });
  });

  it("should contain an image with correct source and class for each element", () => {
    const { getByAltText } = render(tree);

    orgs.map(org => {
      expect(getByAltText(`${org.id} Organization Avatar`)).toHaveProperty(
        "src",
        `http://localhost/${org.avatar_url}`
      );
      expect(getByAltText(`${org.id} Organization Avatar`)).toHaveProperty(
        "className",
        "orgAvatar"
      );

      return false;
    });
  });
});
