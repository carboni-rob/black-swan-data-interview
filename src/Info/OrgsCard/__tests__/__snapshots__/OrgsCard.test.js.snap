// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrgsCard /> should match last snapshot 1`] = `
<div
  class="card"
>
  <h2>
    User Organizations: 
    3
  </h2>
  <div
    class="cardContent"
  >
    <a
      href="test1/orgs/url"
    >
      <img
        alt="1 Organization Avatar"
        class="orgAvatar"
        src="test1/avatar/url"
      />
      test1
    </a>
    <a
      href="test2/orgs/url"
    >
      <img
        alt="2 Organization Avatar"
        class="orgAvatar"
        src="test2/avatar/url"
      />
      test2
    </a>
    <a
      href="test3/orgs/url"
    >
      <img
        alt="3 Organization Avatar"
        class="orgAvatar"
        src="test3/avatar/url"
      />
      test3
    </a>
  </div>
</div>
`;
